src/sentry/api/bases/organization_events.py

```diff
@@ -156,13 +156,15 @@ def get_snuba_query_args_legacy(self, request, organization):
                 'Boolean search operator OR and AND not allowed in this search.')
         return snuba_args
 
+    # TODO(lyn): Deprecate in favour of eventstore.get_next_event_id
```

Comment:
We should avoid having this code mostly duplicated here and in eventstore. The two implementations can diverge before we are done with the migration. Please move the code into eventstore and make this one depend on it. Possibly keep the changes to a minimum, you can evolve it in the next PR.

Reply:
Updated